main.cpp

```diff
@@ -7,42 +7,371 @@
 using namespace std;
 
 class Figure {
-public:
-	Figure() {}
-	virtual bool check_turn() { return true; }
+	int row, col;
+	int row_turn, col_turn;
+	char color;
+public:	
+	Figure(int st_row, int st_col, char st_color) {
+		row = st_row;
+		col = st_col;
+		color = st_color;
+		row_turn = col_turn = 0;
+	}
+	virtual bool check_turn(int dst_row, int dst_col, char dst_color) {
+	/*	if (dst_row < col) col_turn = -col_turn;
+		if (dst_row < row) row_turn = -row_turn;
+		for (int i = col_turn, j = row_turn; col_turn + i < dst_col; i += col_turn, j += row_turn) {
+			if (field[row + j][col + i]->name != "0") return 0;
+		}
+		return 1;*/
+		return true;
+		// TODO: make check based on row_turn, col_turn
+	}
+	char get_color() const {
+		return color;
+	}
+	char get_row() const {
+		return row;
+	}
+	char get_col() const {
+		return col;
+	}
+	void change_row(int row_n) {
```

Comment:
Typically, this kind of function is called setter and thus more conventionally is to have a name "set_row" and "set_col"

Reply:
By the way, it is better to mark comment as resolved after you commit and push changes, not before. It would be a normal commit to just change names of these functions (you can have in commit message something like "Rename change_row/col to set_row/col")